backend-rust/src/indexer.rs

```diff
@@ -4555,20 +4561,29 @@ impl PreparedPaydayCommissionRates {
     }
 }
 
-/// Represents the payday lottery power updates for bakers captured from
+/// Represents the lottery power updates for bakers captured from
 /// the `get_election_info` node endpoint.
-struct PreparedPaydayLotteryPowers {
+struct PreparedLotteryPowers {
     baker_ids:             Vec<i64>,
     bakers_lottery_powers: Vec<BigDecimal>,
+    ranks:                 Vec<i64>,
 }
 
-impl PreparedPaydayLotteryPowers {
+impl PreparedLotteryPowers {
     fn prepare(bakers: Vec<BirkBaker>) -> anyhow::Result<Self> {
         let capacity = bakers.len();
         let mut baker_ids: Vec<i64> = Vec::with_capacity(capacity);
         let mut bakers_lottery_powers: Vec<BigDecimal> = Vec::with_capacity(capacity);
+        let mut ranks: Vec<i64> = Vec::with_capacity(capacity);
+
+        // Sort bakers by lottery power. The baker with the highest lottery power comes
+        // first in the vector and gets rank 1.
+        let mut sorted_bakers: Vec<&BirkBaker> = bakers.iter().collect();
+        sorted_bakers.sort_by(|self_baker, other_baker| {
```

Comment:
Since we take ownership of the `bakers` vec, we can just sort that one directly
```suggestion
    fn prepare(mut bakers: Vec<BirkBaker>) -> anyhow::Result<Self> {
        let capacity = bakers.len();
        let mut baker_ids: Vec<i64> = Vec::with_capacity(capacity);
        let mut bakers_lottery_powers: Vec<BigDecimal> = Vec::with_capacity(capacity);
        let mut ranks: Vec<i64> = Vec::with_capacity(capacity);

        // Sort bakers by lottery power. The baker with the highest lottery power comes
        // first in the vector and gets rank 1.
        bakers.sort_by(|self_baker, other_baker| {
```